symengine/ntheory.cpp

```diff
@@ -1474,4 +1474,32 @@ void powermod_list(std::vector<RCP<const Integer>> &pows, const RCP<const Intege
     }
 }
 
+std::vector<int> quadratic_residue(const Integer &a)
+{
+    /*
+        Returns the list of quadratic residues.
+        Examples
+        ========
+        >>> from sympy.ntheory.residue_ntheory import quadratic_residues
+        >>> quadratic_residues(7)
+        [0, 1, 2, 4]
+    */
+
+    if (a.as_int() < 1) {
+        throw std::runtime_error("quadratic_residue: Input must be > 0");
+    }
+
+    std::vector<int> residue;
+    for( int i = 0; i <= a.as_int()/2; i++)
+    {
+        residue.push_back( (int)pow(i,2) % a.as_int());
```

Comment:
I guess that could be the problem since tests pass on 64 bit version. I'll look into it.